content/docs/contributing/featuregates.md

```diff
@@ -0,0 +1,48 @@
+---
+Title: Implementing feature gates
+description: 'cert-manager contributing guide: Implementing feature gates'
+---
+
+As of v1 release cert-manager is considered stable. We aim to follow Kubernetes API compatibility policy when making API changes, see [API compatibility](../installation/api-compatibility.md) to avoid breaking users' existing cert-manager installations.
+This means that as developers we are somewhat limited in regards to changing existing behavior, i.e renaming or removing API elements or changing their behavior.
+
+
+New functionality that is not yet stable[^1] can still be added, but it needs to be placed behind a feature gate.
+
+## Feature gated API fields
+
+Feature gated API fields are implemented using `--feature-gates` flags of cert-manager [webhook](../cli/webhook.md) and [controller](../cli/controller.md).
```

Comment:
Should they always have the same name? If so, it'd be good to spell that out.

Reply:
This is really up to the implementor